### DISCUSSION TITLE
Adventure map, add a quick choice window for combat mode selection

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#3593] [Ideas]
In fheroes2, it would be nice to be able to choose quickly, directly on the adventure map, the combat mode ("Manual", "Auto Resolve", "Auto, No spells").

Screens are for explanation only.

When the predetermined path of the hero is displayed to attack a wandering troop or a hero, etc., and the mouse cursor is positioned over the target to be attacked (sword cursor), activating the mouse wheel (or pressing the "CTRL" key on the keyboard) brings up a window that allows to choose quickly the desired combat mode.
![Combat mode selection 001](https://user-images.githubusercontent.com/43583994/120497263-8bb48c00-c3be-11eb-8d1c-9169fe5bfbb7.png)

Also when it is possible to attack in the adjacent squares.
![Combat mode selection 001a](https://user-images.githubusercontent.com/43583994/120502316-9c670100-c3c2-11eb-868b-ad3256352a2a.png)

The currently selected mode is indicated by a flashing colored frame (as when selecting a troop in the Hero Meeting screen).
The program remembers the last mode selected by the player.
(The user can also make a double click (or a second click when the mode is already selected) to validate and close the window.)
![Combat mode selection quick window 005](https://user-images.githubusercontent.com/43583994/120497339-996a1180-c3be-11eb-86ab-4fd7ae3f655b.png)

**Comment by Kromster80** [comment]
Perhaps it's even better to just do the combat in Auto mode (without asking anything) and then allow to "re-battle" in the desired mode selected from the battle results window (similar to how HotA allows to re-battle any autocombat in SP mode).

**Reply by Branikolog** [reply]
@Kromster80 
Which is already implemented in fheroes2. ;)

**Reply by Kromster80** [reply]
Is it? I can only select to "Restart" and do the battle manually ..

**Reply by Branikolog** [reply]
Well, maybe I misunderstood your previous statement... Correct me, if I'm wrong: in HoTa we choose normal/instant battle mode in options, but can re-battle after a battle.
The same was implemented in fheroes2 with the only difference: we cannot restart previously played manual battle. Which prevents from spamming the "Restart" button. 
The most comfortable way to play with such options is to choose instant battle with no spells as default. This helps to win the majority of easy battles. For harder ones - manual restart is your choice. ;)

**Reply by Kromster80** [reply]
Okay. Either way, I'm not liking the proposed "quick choice window for combat mode selection" :-)

**Reply by LeHerosInconnu** [reply]
Hello @Kromster80 and @Branikolog,

This is a feature I thought would be interesting to have by default in fheroes2.
But I think I'll make it an option in my mod then. :)
It is part of a series of features aimed at improving the player's gaming experience: https://github.com/ihhub/fheroes2/discussions/3282, https://github.com/ihhub/fheroes2/discussions/3281.

@Branikolog.

> ...we cannot restart previously played manual battle. Which prevents from spamming the "Restart" button.

What's the problem with that?
Let the user play the game the way he wants.
Personally, I like to replay a combat in different ways to see if the result can be improved.
The "RESTART" button would allow this without having to load a save file, thus improving the gaming experience.

@Kromster80.

> Okay. Either way, I'm not liking the proposed "quick choice window for combat mode selection" :-)

You always have the choice to simply ignore it, and nothing will be changed for you. :D

**Comment by LeHerosInconnu** [comment]
Maybe I'll revisit that later.
I close the discussion.
